vmm/src/cpu.rs

```diff
@@ -983,6 +1002,31 @@ impl CpuManager {
         vcpu_thread_barrier: Arc<Barrier>,
         inserting: bool,
     ) -> Result<()> {
+        {
+            #[cfg(feature = "kvm")]
+            {
+                let raw_kvm_fd = vcpu.lock().unwrap().get_kvm_vcpu_raw_fd();
+
+                let buffer = unsafe {
+                    libc::mmap(
+                        core::ptr::null_mut(),
+                        4096,
+                        libc::PROT_WRITE | libc::PROT_READ,
+                        libc::MAP_SHARED,
+                        raw_kvm_fd,
+                        0,
+                    )
+                };
+                assert_ne!(buffer, libc::MAP_FAILED);
+
+                let mut vcpus = STATIC_VCPUS.write().unwrap();
+                vcpus.push((
+                    vcpu.lock().unwrap().id,
+                    SafeWrapper(buffer.cast::<kvm_run>()),
+                ));
```

Comment:
If we use a thread local for kvm_run as I suggested above then this would instead be:
```suggestion
                KVM_RUN.set(buffer.cast::<kvm_run>());

```

EDIT:
We would have to set this inside the newly spawned thread a bit further down in the function though. I don't see any problems with doing that however.